src/core/componentSystems/PhysicsSystem.js

```diff
@@ -58,230 +57,160 @@ export class PhysicsSystem extends MRSystem {
+        entity.physics.collider = this.app.physicsWorld.createCollider(colliderDesc, entity.physics.body);
+        mrjsUtils.physics.COLLIDER_ENTITY_MAP[entity.physics.collider.handle] = entity;
+        entity.physics.collider.setActiveCollisionTypes(mrjsUtils.physics.RAPIER.ActiveCollisionTypes.DEFAULT | mrjsUtils.physics.RAPIER.ActiveCollisionTypes.KINEMATIC_FIXED);
+        entity.physics.collider.setActiveEvents(mrjsUtils.physics.RAPIER.ActiveEvents.COLLISION_EVENTS);
 
-        entity.dispatchEvent(new MouseEvent('mouseleave'));
-    };
+
+    }
 
     /**
      * @function
-     * @description When a new entity is created, adds it to the physics registry and initializes the physics aspects of the entity.
-     * @param {MREntity} entity - the entity being set up
+     * @description Initializes a Rigid Body detailed convexMesh collider for the entity 
+     * NOTE: not currently in use until we can sync it with animations
+     * @param {MREntity} entity - the entity being updated
      */
-    onNewEntity(entity) {
-        this.initPhysicsBody(entity);
-        this.registry.add(entity);
+    initDetailedBody(entity) {
+    
+        const rigidBodyDesc = mrjsUtils.physics.RAPIER.RigidBodyDesc.fixed()
+        entity.physics.body = this.app.physicsWorld.createRigidBody(rigidBodyDesc);
+
+        entity.physics.colliders = []
+
+        entity.object3D.traverse((child) => {
+            if(child.isMesh) {
+                let collider = this.app.physicsWorld.createCollider(this.initConvexMeshCollider(child, entity.compStyle.scale), entity.physics.body)
+                collider.setCollisionGroups(mrjsUtils.physics.CollisionGroups.UI);
+                entity.physics.colliders.push(collider)
+                mrjsUtils.physics.COLLIDER_ENTITY_MAP[collider.handle] = entity;
+                collider.setActiveCollisionTypes(mrjsUtils.physics.RAPIER.ActiveCollisionTypes.DEFAULT | mrjsUtils.physics.RAPIER.ActiveCollisionTypes.KINEMATIC_FIXED);
+                collider.setActiveEvents(mrjsUtils.physics.RAPIER.ActiveEvents.COLLISION_EVENTS);
+
+            }
+        })
+
     }
 
-    /**
+     /**
      * @function
-     * @description Initializes the rigid body used by the physics part of the entity
+     * @description Initializes a convexMesh collider from a THREE.js geometry
+     * NOTE: not currently in use until we can sync it with animations
      * @param {MREntity} entity - the entity being updated
      */
-    initPhysicsBody(entity) {
-        if (entity.physics.type == 'none') {
-            return;
+    initConvexMeshCollider(object3D, scale) {
+        const positionAttribute = object3D.geometry.getAttribute('position');
+        const vertices = [];
+        for (let i = 0; i < positionAttribute.count; i++) {
+            const vertex = new THREE.Vector3().fromBufferAttribute(positionAttribute, i).multiplyScalar(scale).multiplyScalar(mrjsUtils.app.scale);
+            vertices.push([vertex.x, vertex.y, vertex.z]);
```

Comment:
it's better to use threejs's `toArray` function here instead of [vertex.x, vertex.y, vertex.z]

Reply:
also `multiplyScalar(scale).multiplyScalar(mrjsUtils.app.scale);` can be shortened to `multiplyScalar(scale * mrjsUtils.app.scale)`